lib/zold/version_upgrade_manager.rb

```diff
@@ -0,0 +1,48 @@
+require 'semantic'
+
+module Zold
+  # Class to manage data upgrades (when zold itself upgrades).
+  class VersionUpgradeManager
+    # @todo #285:30min Refactor so we don't just assume there's .zoldata
+    #  in the current directory. Possibly something like a config object
+    #  should be introduced.
+    #
+    # @todo #285:30min Communicate the ongoing upgrade and its steps to the user.
+    #  The user should know:
+    #  - If the upgrade is happening.
+    #  - What upgrade scripts are run.
+    #  - On failure there should be a report of what failed.
+    #
+    # @todo #285:30min Write the upgrade manager tests that ensure:
+    #  - Nothing breaks without the version file.
+    #  - The upgrade scripts run when there is a version file and there are pending upgrade scripts.
+    #  - Make sure *only* the correct upgrade scripts run.
+    def self.run
+      from_version = data_version
+      return unless from_version
+
+      upgrade(from_version)
+      File.open('.zoldata/version', 'w') do |file|
+        file.puts(Zold::VERSION)
+      end
+    end
+
+    def self.data_version
+      version_string = File.read('.zoldata/version')
+      Semantic::Version.new(version_string)
+    rescue Errno::ENOENT
```

Comment:
@botanicus this looks like using exceptions for flow control, which is an anti-pattern: https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why